api/v1alpha1/instaslice_types.go

```diff
@@ -36,6 +37,61 @@ const (
 	AllocationStatusCreated  AllocationStatusDaemonset  = "created"
 )
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// InstasliceOperator is the Schema for the InstasliceOperator API
+// +k8s:openapi-gen=true
+// +genclient
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+type InstasliceOperator struct {
+	// TypeMeta data for the operator
+	// +required
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata for the operator
+	// +required
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec InstasliceOperatorSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status InstasliceOperatorStatus `json:"status"`
+}
+
+// InstasliceOperatorSpec defines the desired state of InstasliceOperator
+type InstasliceOperatorSpec struct {
```

Comment:
@rphillips Maybe we can add a field for specifying operator wide log level as well.